vyxal/elements.py

```diff
@@ -1739,12 +1739,15 @@ def gen_from_fn(lhs, rhs, ctx):
     def gen():
         yield from lhs
 
-        made = lhs
+        made = [*lhs]
 
+        func_arity = (
+            rhs.stored_arity if "stored_arity" in dir(rhs) else rhs.arity
+        )
```

Comment:
Do we want to make a helper for this? I feel like I've seen a similar snippet elsewhere.

Reply:
You've seen a similar snippet in the transpilation for lambdas.